ArchiSteamFarm/ArchiWebHandler.cs

```diff
@@ -163,33 +176,27 @@ public sealed class ArchiWebHandler : IDisposable {
 					Steam.InventoryResponse response = await UrlGetToJsonObjectWithSession<Steam.InventoryResponse>(SteamCommunityURL, request + (startAssetID > 0 ? "&start_assetid=" + startAssetID : "")).ConfigureAwait(false);
 
 					if (response == null) {
-						return null;
+						throw new Exception(string.Format(Strings.ErrorObjectIsNull, nameof(response)));
 					}
 
 					if (!response.Success) {
-						Bot.ArchiLogger.LogGenericWarning(!string.IsNullOrEmpty(response.Error) ? string.Format(Strings.WarningFailedWithError, response.Error) : Strings.WarningFailed);
-
-						return null;
+						throw new Exception(!string.IsNullOrEmpty(response.Error) ? string.Format(Strings.WarningFailedWithError, response.Error) : Strings.WarningFailed);
 					}
 
 					if (response.TotalInventoryCount == 0) {
 						// Empty inventory
-						return result.Values.ToHashSet();
+						yield break;
 					}
 
 					if ((response.Assets == null) || (response.Assets.Count == 0) || (response.Descriptions == null) || (response.Descriptions.Count == 0)) {
-						Bot.ArchiLogger.LogNullError(nameof(response.Assets) + " || " + nameof(response.Descriptions));
-
-						return null;
+						throw new Exception(string.Format(Strings.ErrorObjectIsNull, nameof(response.Assets) + " || " + nameof(response.Descriptions)));
```

Comment:
Exactly because stuff like that shouldn't be handled and silently discarded, this should be `ArgumentException`.

Then in general flow, there should be:
- catching IOException as network failure, without logging (which is done by WebBrowser already), just return appropriate failure to user.
- catching Exception (including ArgumentException) and logging as generic exception through ArchiLogger.

Reply:
This way we keep previous flow, where `IOException` will be handled same as `null` inventory, while serious stuff will be logged like before, just not here but by the caller.